pkg/server/server.go

```diff
@@ -251,7 +252,28 @@ func (s *Server) Close() error {
 	return errors.Collect(ErrCloseServer, errs...)
 }
 
-func buildEtcd(cfg *config.Config, logger *zap.Logger, pubAddr string, engine *gin.Engine) (srv *embed.Etcd, err error) {
+func buildEtcd(cfg *config.Config, logger *zap.Logger, engine *gin.Engine) (srv *embed.Etcd, err error) {
+	if cfg.Cluster.ClusterName == "" {
+		return nil, errors.New("cluster_name can not be empty")
+	}
+	if cfg.Cluster.NodeName == "" {
+		cfg.Cluster.NodeName = fmt.Sprintf("%s-%d", cfg.Cluster.ClusterName, time.Now().UnixMicro())
```

Comment:
Does PD also generate it like this?

Reply:
No, but in a smilar fashion. It needs to be unique in the cluster.

Reply:
Can we also generate it like that? A hostname is better than a random timestamp.